assets/js/modules/analytics/util/convertDateStringToDateObject.test.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * convertDateStringToDateObject utility tests
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import convertDateStringToDateObject from './convertDateStringToDateObject';
+
+describe( 'convertDateStringToDateObject', () => {
+	it( 'converts a date string formatted as YYYYMMDD', () => {
+		const dateToConvert = '20200608';
+		const convertedDate = convertDateStringToDateObject( dateToConvert );
+		expect( convertedDate.getFullYear() ).toBe( 2020 );
+		expect( convertedDate.getMonth() ).toBe( 5 ); // Months are zero-indexed.
+		expect( convertedDate.getDate() ).toBe( 8 );
+	} );
+	it( 'returns false when passed an incorrect date', () => {
+		const dateToConvert = 'notadate';
+		const convertedDate = convertDateStringToDateObject( dateToConvert );
+		expect( convertedDate ).toBe( false );
+	} );
```

Comment:
Let's add a few tests here for non-string types to make sure those are covered as well 👍